desmume/src/frontend/modules/osd/agg/aggdraw.cpp

```diff
@@ -162,6 +162,23 @@ void AggDraw_Desmume::setTarget(AggTarget newTarget)
 	target = targets[newTarget];
 }
 
+void Agg_setCustomSize(int w, int h)
+{
+	hudBuffer.resize(w*h);
+	luaBuffer.resize(w*h);
+	if(aggDraw.screen)
+	{
+#if defined(WIN32) || defined(HOST_LINUX)
+		aggDraw.screen->setDrawTargetDims(0, w, h, w*4);
+#else
+		aggDraw.screen->setDrawTargetDims(0, w, h, w*2);
```

Comment:
what's the difference between linux/win and everything else here ?

Reply:
See also line 121, was not added by me, I only mimiced this in Agg_setCustomSize. If neither WIN32 nor HOST_LINUX is defined than screen buffer is for some reason assumed to be 16bit, which causes HUD to be drawn incorrectly. I don't know what original author ment by this, for which device. May be it's better to use define to specify buffer bitness instead of checking for some platforms?

Reply:
sounds reasonable. we can store byteperpixel used to initialize SDL2, and use it for math rather than hardcoding numbers nobody knows what they mean.

Reply:
Added bytes per pixel detection, defaults to 4 (at start and in case SDL call fails for some reason).